dockerfiles/csharp/Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM golang:1.10-alpine
+WORKDIR /go/src/github.com/sourcegraph/lsp-adapter
+COPY . .
+RUN CGO_ENABLED=0 GOBIN=/usr/local/bin go install github.com/sourcegraph/lsp-adapter
+
+# .NET Core 2.0.0 is not available on Alpine yet, but it is actively being
+# worked on: https://github.com/dotnet/core/issues/1076
+FROM ubuntu:16.04
+
+RUN apt-get update \
+  && apt-get install -y gnupg curl \
+  # Ubuntu 16.04's default nodejs package v4.2.6 is too old to run the
+  # omnisharp-client npm package.
+  && curl -sL https://deb.nodesource.com/setup_10.x | bash - \
+  && curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > /etc/apt/trusted.gpg.d/microsoft.gpg \
+  && echo "deb [arch=amd64] https://packages.microsoft.com/repos/microsoft-ubuntu-xenial-prod xenial main" > /etc/apt/sources.list.d/dotnetdev.list \
+  && echo "deb https://download.mono-project.com/repo/ubuntu stable-xenial main" > /etc/apt/sources.list.d/mono-official-stable.list \
+  && apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF \
+  && apt-get update && apt-get install -y apt-transport-https \
+  # OmniSharp runs on Mono and requires the .NET SDK 2.0.0.
+  && apt-get update && apt-get install -y mono-complete dotnet-sdk-2.0.0 nodejs \
+  # This is the language server itself. Technically, it's a wrapper which lazily
```

Comment:
what triggers the lazy download? can't we do that at build time of the Dockerfile?

Reply:
https://github.com/OmniSharp/omnisharp-node-client/blob/fcb57e1f5c9d4f39559499861fd5bc031964b787/lib/drivers/StdioDriver.ts#L102

I'll see about baking it into the image.

Reply:
I managed to bake it into the image :zap: